marine/src/config/marine_config.rs

```diff
@@ -172,11 +172,14 @@ impl TryFrom<TomlMarineConfig> for MarineConfig {
 
     fn try_from(toml_config: TomlMarineConfig) -> Result<Self, Self::Error> {
         let base_path = PathBuf::from(toml_config.base_path);
-        let context = ConfigContext { base_path };
+        let context = ConfigContext {
+            base_path: Some(base_path),
+        };
+        // unwrap is safe because it was set to Some in the line above;
 
         let modules_dir = toml_config
             .modules_dir
-            .map(|dir| adjust_path(&context.base_path, Path::new(&dir)))
+            .map(|dir| as_relative_to_base(context.base_path.as_ref(), Path::new(&dir)))
```

Comment:
I guess you could simplify to
```
.map(|dir| as_relative_to_base(&context.base_path, &dir))
```
but I'm not 100% sure

Reply:
`dir` is a `String`, so i have to change it to PatBuf in the config. Maybe its even better.